APMrover2/AP_MotorsUGV.cpp

```diff
@@ -591,6 +608,63 @@ void AP_MotorsUGV::output_skid_steering(bool armed, float steering, float thrott
     output_throttle(SRV_Channel::k_throttleRight, 100.0f * motor_right);
 }
 
+// output for custom configurations
+void AP_MotorsUGV::output_custom_config(bool armed, float steering, float throttle, float lateral)
+{
+    // exit immediately if the frame type is set to UNDEFINED
+    if (FRAME_TYPE_UNDEFINED) {
+        return;
+    }
+
+    if (armed) {
+
+        // clear and set limits based on input
+        set_limits_from_input(armed, steering, throttle);
+
+        // constrain steering
+        steering = constrain_float(steering, -4500.0f, 4500.0f);
+
+        // reverse steering when throttle is negative
+        if (is_negative(throttle)) {
+            steering *= -1.0f;
+        }
+
+        // scale throttle, steering and lateral inputs to -1 to 1
+        float scaled_throttle = throttle / 100.0f;
+        float scaled_steering = steering / 4500.0f;
+        float scaled_lateral = lateral / 100.0f;
+
+        float thr_str_ltr_out[max_num_motors];
```

Comment:
those are only use in the for loop, so no need of a table ... just reduce the scope